### PR TITLE
mediatek: filogic: add Qihoo 360T7 support

### PR DESCRIPTION
**Pull request by aiamadeus**
Hardware specification:
  SoC: MediaTek MT7981B 2x A53
  Flash: ESMT F50L1G41LB 128MB
  RAM: MT5CC128M16JR-EK 256MB
  Ethernet: 4x 10/100/1000 Mbps
  Switch: MediaTek MT7531AE
  WiFi: MediaTek MT7976C
  Button: Reset, WPS
  Power: DC 12V 1A

Flash instructions:
  1. Attach UART, boot the stock firmware until
     the message about failsafe mode appears.
  2. Enter failsafe mode by pressing "f" and "Enter"
  3. Type "mount_root", then run
     "fw_setenv bootmenu_delay 3"
  4. Back up all mtd partitions before flashing.
  5. Reboot, U-Boot now presents a menu.
  6. Connect to your PC via the Gigabit port of the router,
     set a static ip on the ethernet interface of your PC.
     (ip 192.168.1.254, gateway 192.168.1.1)
  7. Select "Upgrade ATF BL2", then use this file:
     openwrt-mediatek-filogic-qihoo_360t7-preloader.bin
  8. Select "Upgrade ATF FIP", then use this file:
     openwrt-mediatek-filogic-qihoo_360t7-bl31-uboot.fip
  9. Download the initramfs image, and type "reset",
     waiting for tftp recovery to complete.
  a. After openwrt boots up, perform sysupgrade.

Note:
  1. Since NMBM is disabled, we must back up all partitions.
  2. Flash instructions is based on commit https://github.com/openwrt/openwrt/commit/28df7f7ff251d8684caa60f07ed0ba0c4a8e71f4.

### CODE REVIEW

**Comment by dangowrt** [comment]
@aiamadeus Does WED work as expected if enabled permantly via `/etc/modules.conf`?

File: target/linux/mediatek/dts/mt7981b-qihoo-360t7.dts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		port@1 {
+			reg = <1>;
+			label = "lan2";
+		};
+
+		port@2 {
+			reg = <2>;
+			label = "lan1";
+		};
+
```
Comment (cwbsw) [inline target/linux/mediatek/dts/mt7981b-qihoo-360t7.dts:176]: I suggest label port0 as lan1, label port2 as lan3, to keep compatible with existing firmware.

File: package/boot/arm-trusted-firmware-mediatek/Makefile [hunk 2]
```diff
@@ -197,6 +243,7 @@ TFA_MAKE_FLAGS += \
 	BOOT_DEVICE=$(BOOT_DEVICE) \
 	USE_MKIMAGE=1 MKIMAGE=$(STAGING_DIR_HOST)/bin/mkimage \
 	$(if $(findstring ddr4,$(DDR_TYPE)),DRAM_USE_DDR4=1) \
+	$(if $(BOARD_QFN),BOARD_QFN=1,BOARD_BGA=1) \
```
[the remark this answers is not in the document]
Comment (aiamadeus) [reply package/boot/arm-trusted-firmware-mediatek/Makefile:246]: https://github.com/mtk-openwrt/arm-trusted-firmware/blob/mtksoc/plat/mediatek/mt7981/platform.mk#L98
Maybe I understand wrong?

File: target/linux/mediatek/dts/mt7981b-qihoo-360t7.dts (hunk 1 above)
Comment (aiamadeus) [reply target/linux/mediatek/dts/mt7981b-qihoo-360t7.dts:176]: The origin firmware uses the switch driver. In dsa driver, the port name should be consistent with the shell label.

Comment (cwbsw) [reply target/linux/mediatek/dts/mt7981b-qihoo-360t7.dts:176]: > The stock firmware uses the switch driver. In dsa driver, the port name should be consistent with the shell label.

In fact, there is no label on the shell. It's better label port0 as lan1 in order to follow the reading habits of most people.

File: package/boot/arm-trusted-firmware-mediatek/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (dangowrt) [reply package/boot/arm-trusted-firmware-mediatek/Makefile:246]: Looks like you are right, we do have them both there.

**Comment by dangowrt** [comment]
Picked to main branch with small fix applied.

**Pull request closed without merging**